common/src/service/HttpClientHelper.cs

```diff
@@ -1,56 +1,84 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace Microsoft.Azure.Devices
-{
-    using Microsoft.Azure.Devices.Common;
-    using Microsoft.Azure.Devices.Common.Exceptions;
-    using Microsoft.Azure.Devices.Common.Extensions;
-    using Microsoft.Azure.Devices.Shared;
-    using System;
-    using System.Collections.Generic;
-    using System.Collections.ObjectModel;
-    using System.Globalization;
-    using System.IO;
-    using System.Linq;
-    using System.Net;
-    using System.Net.Http;
-    using System.Net.Http.Headers;
-    using System.Threading;
-    using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Common.Extensions;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
 #if NET451
     using System.Net.Http.Formatting;
 #endif
 
-    sealed class HttpClientHelper : IHttpClientHelper
+namespace Microsoft.Azure.Devices
+{
+    internal sealed class HttpClientHelper : IHttpClientHelper
     {
 #if !NETSTANDARD1_3 && !NETSTANDARD2_0
         static readonly JsonMediaTypeFormatter JsonFormatter = new JsonMediaTypeFormatter();
 #endif
-        readonly Uri baseAddress;
-        readonly IAuthorizationHeaderProvider authenticationHeaderProvider;
-        readonly IReadOnlyDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> defaultErrorMapping;
-        readonly TimeSpan defaultOperationTimeout;
-        readonly IWebProxy customHttpProxy;
-        readonly Action<HttpClient> preRequestActionForAllRequests;
+        private readonly Uri _baseAddress;
+        private readonly IAuthorizationHeaderProvider _authenticationHeaderProvider;
+        private readonly IReadOnlyDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> _defaultErrorMapping;
+        private readonly TimeSpan _defaultOperationTimeout;
 
+        // IDisposables
+
+        private readonly HttpClient _httpClientWithDefaultTimeout;
```

Comment:
They key part of the fix is caching the only 2 instances of HttpClient we need, rather than building a new one for each request.